### PR TITLE
[ncl] Add light sensor

### PR DESCRIPTION
**Pull request by barthap**
# Why

Follow up #18225 

The NCL screen didn't include the new sensor, also crashed on iOS.

# How

- Added "Light Sensor" pane to NCL Sensors screen
- Created `ExpoLightSensor.ios.ts` module stub, so calling the module method `isAvailableAsync` doesn't crash because the module doesn't exist in the registry.
  > I'm wondering if its worth implementing this on the native side 🙄 I doubt that iOS gets light sensor anytime soon and we might want to rewrite the whole expo-sensors anyway.

# Test Plan

NCL on Android and iOS

### CODE REVIEW

**Comment by fobos531** [comment]
> I'm wondering if its worth implementing this on the native side 🙄 I doubt that iOS gets light sensor anytime soon and we might want to rewrite the whole expo-sensors anyway.

Hey @barthap ! I might be wrong, but it seems that Apple provides that information through `ambientLightSensor` in their SensorKit framework. Though, SensorKit is only available on iOS14+ while Expo is still supporting iOS 12/13. And, as you mentioned, if this sensor were to be implemented, I assume it would be a good idea to explore using SensorKit entirely for the rewrite of the `expo-sensors` module.

**Comment by barthap** [comment]
Thanks @fobos531 for the information!
It is true that SensorKit supports ambient light, but from [what I read in their documentation](https://developer.apple.com/documentation/sensorkit/configuring_your_project_for_sensor_reading?language=objc), they require some custom manually-given entitlements in order to use it. So this is definitely not an option for Expo Go.
I mentioned the issue in this discussion: https://github.com/expo/expo/discussions/18101#discussioncomment-3086198

File: apps/native-component-list/src/screens/SensorScreen.tsx [hunk 1]
```diff
@@ -180,6 +181,16 @@ class BarometerSensor extends SensorBlock<Sensors.BarometerMeasurement> {
   );
 }
 
+class LightSensor extends SensorBlock<Sensors.LightSensorMeasurement> {
+  getName = () => 'LightSensor';
+  getSensor = () => Sensors.LightSensor;
```
Comment (tsapeta) [inline apps/native-component-list/src/screens/SensorScreen.tsx:186]: Is this going to work on iOS without errors? Shouldn't we render this component only on Android?

Comment (barthap) [reply apps/native-component-list/src/screens/SensorScreen.tsx:186]: The sensors NCL screen loads only sensors for which `isAvailableAsync() == true`
So it isn't displayed on iOS, web and some Android phones without such sensor (tested it)

**Review by tsapeta — APPROVED** [review]

**Pull request merged**